src/loader/m3u8-parser.ts

```diff
@@ -111,51 +108,54 @@ export default class M3U8Parser {
     return levels;
   }
 
-  static parseMasterPlaylistMedia (string, baseurl, type, audioGroups = []) {
-    let result;
-    let medias = [];
+  static parseMasterPlaylistMedia (string: string, baseurl: string, type: MediaPlaylistType, audioGroups: Array<AudioGroup> = []): Array<MediaPlaylist> {
+    let result: RegExpExecArray | null;
+    let medias: Array<MediaPlaylist> = [];
     let id = 0;
     MASTER_PLAYLIST_MEDIA_REGEX.lastIndex = 0;
     while ((result = MASTER_PLAYLIST_MEDIA_REGEX.exec(string)) !== null) {
-      const media = {};
       const attrs = new AttrList(result[1]);
       if (attrs.TYPE === type) {
-        media.groupId = attrs['GROUP-ID'];
-        media.name = attrs.NAME;
-        media.type = type;
-        media.default = (attrs.DEFAULT === 'YES');
-        media.autoselect = (attrs.AUTOSELECT === 'YES');
-        media.forced = (attrs.FORCED === 'YES');
+        const media: MediaPlaylist = {
+          id: id++,
+          groupId: attrs['GROUP-ID'],
+          name: attrs.NAME ? attrs.NAME : attrs.LANGUAGE,
```

Comment:
```suggestion
          name: attrs.NAME || attrs.LANGUAGE,
```

up to you. not sure if we use this shortcut in other places